indico/modules/events/registration/forms.py

```diff
@@ -423,3 +424,33 @@ class RejectRegistrantsForm(IndicoForm):
 
     def is_submitted(self):
         return super().is_submitted() and 'submitted' in request.form
+
+
+class RegistrationTagForm(IndicoForm):
+    """Form to create a new registration tag."""
+
+    title = StringField(_('Title'), [DataRequired()])
+    color = StringField(_('Color'), [DataRequired()], widget=ColorInput())
+
+    def __init__(self, *args, **kwargs):
+        self.event = kwargs.pop('event', None)
+        super().__init__(*args, **kwargs)
+
+    def validate_title(self, field):
+        query = RegistrationTag.query.with_parent(self.event).filter(
+            db.func.lower(RegistrationTag.title) == field.data.lower()
+        )
+        if query.has_rows():
+            raise ValidationError(_('This title is already in use.'))
+
+
+class RegistrationTagsAssignForm(IndicoForm):
+    """Form to assign registration tags to registrations."""
+
+    add = IndicoMultipleTagSelectField(_('Add'), description=_('Select tags to assign'))
+    remove = IndicoMultipleTagSelectField(_('Remove'), description=_('Select tags to remove'))
+    registration_id = HiddenFieldList()
+    submitted = HiddenField()
+
```

Comment:
Might be nice to fail validation when selecting a tag both for adding anad removal:

```python
    def validate_remove(self, field):
        if set(self.remove.data) & set(self.add.data):
            raise ValidationError(_('You cannot add and remove the same tag'))

    validate_add = validate_remove
```